hummingbot/connector/parrot.py

```diff
@@ -0,0 +1,94 @@
+import aiohttp
+import asyncio
+from typing import List, Dict
+from dataclasses import dataclass
+from decimal import Decimal
+import logging
+from hummingbot.connector.exchange.binance.binance_utils import convert_from_exchange_trading_pair
+from hummingbot.core.utils.async_utils import safe_gather
+
+PARROT_MINER_BASE_URL = "https://papi.hummingbot.io/v1/mining_data/"
+
+s_decimal_0 = Decimal("0")
+
+
+@dataclass
+class CampaignSummary:
+    market_id: int = 0
+    trading_pair: str = ""
+    exchange_name: str = 0
+    spread_max: Decimal = s_decimal_0
+    payout_asset: str = ""
+    liquidity: Decimal = s_decimal_0
+    liquidity_usd: Decimal = s_decimal_0
+    active_bots: int = 0
+    reward_per_wk: Decimal = s_decimal_0
+    apy: Decimal = s_decimal_0
+
+
+async def get_campaign_summary(exchange: str, trading_pairs: List[str] = []) -> Dict[str, CampaignSummary]:
+    results = {}
+    try:
+        campaigns = await get_active_campaigns(exchange, trading_pairs)
+        tasks = [get_market_snapshots(m_id) for m_id in campaigns]
+        snapshots = await safe_gather(*tasks, return_exceptions=True)
+        for snapshot in snapshots:
+            if isinstance(snapshot, Exception):
+                raise snapshot
+            if snapshot["items"]:
+                snapshot = snapshot["items"][0]
+                market_id = int(snapshot["market_id"])
+                campaign = campaigns[market_id]
+                campaign.apy = Decimal(snapshot["annualized_return"])
+                oov = snapshot["summary_stats"]["open_volume"]
+                campaign.liquidity = Decimal(oov["oov_eligible_ask"]) + Decimal(oov["oov_eligible_bid"])
+                campaign.liquidity_usd = campaign.liquidity * Decimal(oov["base_asset_usd_rate"])
+                campaign.active_bots = int(oov["bots"])
+        results = {c.trading_pair: c for c in campaigns.values()}
+    except asyncio.CancelledError:
+        raise
+    except Exception:
+        logging.getLogger(__name__).error("Unexpected error while requesting data from Hummingbot API.", exc_info=True)
+    return results
+
+
+async def get_market_snapshots(market_id: int):
+    async with aiohttp.ClientSession() as client:
+        url = f"{PARROT_MINER_BASE_URL}market_snapshots/{market_id}?aggregate=1m"
```

Comment:
I was thinking if we should the process of how we create new exchange connectors; with their own `*_constants.py` for stuff like URLs.

I suppose this is special, considering that `parrot.py` functions more like a `util.py`

Reply:
Yeah, it is still small for now, we could refactor this into its own folder and more specific files later once we know more on what we need from this connector.